### ISSUE TITLE
Enable Link-Time Optimization (LTO)

### ISSUE THREAD

**Issue by zamazan4ik** [thecoshman/http#165]
Hi!

I noticed that in the `Cargo.toml` file Link-Time Optimization (LTO) for the project is not enabled. I suggest switching it on since it will reduce the binary size (always a good thing to have) and will likely improve the application's performance a bit.

I suggest enabling LTO only for the Release builds so as not to sacrifice the developers' experience while working on the project since LTO consumes an additional amount of time to finish the compilation routine. If you think that a regular Release build should not be affected by such a change as well, then I suggest adding an additional `dist` or `release-lto` profile where additionally to regular `release` optimizations LTO will also be added. Such a change simplifies life for maintainers and others interested in the project persons who want to build the most performant version of the application. Using ThinLTO should also help to reduce the build-time overhead with LTO. If we enable it on the Cargo profile level, users, who install the application with `cargo install`, will get the LTO-optimized version "automatically". E.g., check `cargo-outdated` Release [profile](https://github.com/kbknapp/cargo-outdated/blob/master/Cargo.toml#L48).

Basically, it can be enabled with the following lines:

```
[profile.release]
lto = true
```

Thank you.

P.S. It's more like an improvement idea rather than a bug. I created the issue just because the Discussions are disabled for the repo for now.

**Comment by nabijaczleweli** [comment]
This, of course, should be configured at the call-site, but "what if someone actually designed Cargo" is not for here.

Of course, for optimal codegen you want(ed) something closer to https://github.com/systemd/zram-generator/blob/766bfb0d297fe083cc29ab1886d8feff24f2ee58/Cargo.toml#L29
And rust LTO (used to?) straight-up break, like, dwz.
But "what if someone actually designed Cargo" is not for here.

Before:
```
$ l ~/.cargo/bin/http
-rwxr-xr-x 1 nabijaczleweli users 45.5M 09-23 22:34 /home/nabijaczleweli/.cargo/bin/http
$ cp ~/.cargo/bin/http .
$ strip http
$ l http
-rwxr-xr-x 1 nabijaczleweli users 3.8M 10-09 18:58 http
$ size -G ~/.cargo/bin/http http
      text       data        bss      total filename
   2168335    1825561       9152    4003048 /home/nabijaczleweli/.cargo/bin/http
   2168335    1825561       9152    4003048 http
```

After:
```
$ l ~/.cargo/bin/http
-rwxr-xr-x 1 nabijaczleweli users 29M 10-09 19:02 /home/nabijaczleweli/.cargo/bin/http
$ cp ~/.cargo/bin/http .
$ strip http
$ l http
-rwxr-xr-x 1 nabijaczleweli users 3.6M 10-09 19:02 http
$ size -G ~/.cargo/bin/http http
      text       data        bss      total filename
   2120479    1662959       9184    3792622 /home/nabijaczleweli/.cargo/bin/http
   2120479    1662959       9184    3792622 http
```

-48k of .text

eh, sure

**Comment by nabijaczleweli** [comment]
Released in [`v2.2.0`](https://github.com/thecoshman/http/releases/tag/v2.2.0)
